vm/src/org/mozartoz/truffle/nodes/call/CallMethodNode.java

```diff
@@ -33,11 +43,36 @@ public static CallMethodNode create() {
 
 	static final RecordFactory OTHERWISE_MESSAGE_FACTORY = Arity.build("otherwise", 1L).createFactory();
 
-	@Specialization
+	@Specialization(guards = {
+			"cachedProc != null",
```

Comment:
Just my habit, but I put the guards purely on cached values at the end, leaving the runtime guards first.